keras_hub/src/models/gemma/gemma_attention.py

```diff
@@ -130,14 +130,23 @@ def _compute_attention(
             if attention_mask is not None:
                 attention_mask = ops.expand_dims(attention_mask, axis=1)
                 attention_mask = ops.cast(attention_mask, dtype="bool")
-
-            attention_output = ops.dot_product_attention(
-                query=q,
-                key=k,
-                value=v,
-                mask=attention_mask,
-                scale=query_normalization,
-            )
+            if running_on_tpu():
+                attention_output = ops.dot_product_attention(
+                    query=q,
+                    key=k,
+                    value=v,
+                    mask=attention_mask,
+                    scale=query_normalization,
+                    attn_logits_soft_cap=self.logit_soft_cap,
```

Comment:
I don't get it. Does this mean we not using a `logit_soft_cap` when running on GPU? Wouldn't any difference in numerics between TPU and GPU indicate a bug?

Reply:
yes, this is not supported in GPU - it would need to be implemented explicitly. Do we just enable this on TPU and add the fallback option for CPU and GPU?

Reply:
This new version looks broken. `attention_output` is not going to be defined for if `running_on_tpu` is false. Won't this just crash?

Reply:
updated the fallback logic